### PR TITLE
Ci server changes

### PR DESCRIPTION
**Pull request by howarddierking**
One additional question before accepting. If the grunt-based build approach works for us, is there a reason that we should keep the gulp configuration file around? There's certainly no harm in doing so - just wondering if it makes sense to do so.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# http://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+end_of_line = lf
+insert_final_newline = true
+indent_style = space
```
Comment (Trudeaucj) [inline .editorconfig]: :+1:

File: build/record_version.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#! /bin/bash
+
+# only execute this if the branch is dev and it's not a pull request to the branch
+if [ $PULL_REQUEST == 'false' ]
+then
+  # update the package verion and commit to the git repository
+  npm version patch -m "Package version %s created from build $BUILD_NUMBER [ci skip]"
+
+  # use the key that Shippable uses to connect to GitHub
+  ssh-agent bash -c "ssh-add /home/shippable/.ssh/project_rsa; git push --tags origin $BRANCH"
```
Comment (Trudeaucj) [inline build/record_version.sh]: Can we do npm publish as well?

File: package.json [hunk 3]
```diff
@@ -26,13 +26,20 @@
     "xml2js": "^0.4.4"
   },
   "devDependencies": {
+    "mocha": "^1.21.4",
     "chai": "^1.9.1",
     "config": "^1.0.2",
     "gulp": "^3.8.9",
```
Comment (Trudeaucj) [inline package.json]: @howarddierking feel free to remove the gulp file with grunt additions.

File: build/record_version.sh (hunk 2 above)
Comment (howarddierking) [inline build/record_version.sh]: yup - will add this just as soon as I get NPM Enterprise working

File: package.json (hunk 3 above)
Comment (howarddierking) [inline package.json]: done

**Comment by howarddierking** [comment]
ready for review

File: package.json (hunk 3 above)
Comment (Trudeaucj) [inline package.json]: :+1:

**Pull request merged**